### PR TITLE
[Behat][Promotion] Discount based on cart quantity

### PR DESCRIPTION
**Pull request by Zales0123**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets |
| License       | MIT
| Doc PR        |

Based on https://github.com/Sylius/Sylius/pull/4249.
There is few things done in this PR connected with implementation of ``receiving_discount_based_on_cart_quantity`` feature:
- implementation of custom ``RuleFactory`` - as it have been done with ``ActionFactory`` but already in proper namespace
- added new promotions-checking step definition - changed ``my cart promotions`` to ``my discount``, changes in other scenarios will be done in separate PR
- set ``CartQuantityRuleChecker`` to always check ``>=`` condition - this change should be done for all checkers and it will be proposed and explained in separate PR

### CODE REVIEW

File: features/promotion/receiving_discount_based_on_cart_quantity.feature [hunk 1]
```diff
@@ -5,36 +5,36 @@ Feature: Receiving discount based on cart quantity
...
-        Given the store is operating on a single "France" channel
+        Given the store operates on a single channel in "France"
         And the store has a product "PHP T-Shirt" priced at "€100.00"
         And there is a promotion "Holiday promotion"
 
-    @todo
+    @ui
     Scenario: Receiving discount when buying more than required quantity
-        Given the promotion gives "€10.00" fixed discount to every cart with quantity at least 3
+        Given the promotion gives "€10.00" fixed discount to every order with quantity at least 3
```
Comment (michalmarcinkowski) [inline features/promotion/receiving_discount_based_on_cart_quantity.feature]: `cart` is correct, why changing this?

File: features/promotion/receiving_discount_based_on_cart_quantity.feature [hunk 2]
```diff
@@ -5,36 +5,36 @@ Feature: Receiving discount based on cart quantity
...
 
-    @todo
+    @ui
     Scenario: Not receiving discount when buying less than required quantity
-        Given the promotion gives "€10.00" fixed discount to every cart with quantity at least 3
+        Given the promotion gives "€10.00" fixed discount to every order with quantity at least 5
         When I add 2 products "PHP T-Shirt" to the cart
         Then my cart total should be "€200.00"
-        And my discount should be "€0.00"
+        And there should be no discount
```
Comment (michalmarcinkowski) [inline features/promotion/receiving_discount_based_on_cart_quantity.feature]: :+1:

**Comment by pjedrzejewski** [comment]
Requires rebasing. Thanks!

File: features/promotion/receiving_discount_based_on_cart_quantity.feature (hunk 1 above)
Comment (Zales0123) [inline features/promotion/receiving_discount_based_on_cart_quantity.feature]: I've tried to be consistent with previously implemented steps. There is already step "And it gives "€10.00" fixed discount **to every order**". Do you think ``cart`` is more correct and it should be changed in previous steps?

**Comment by pjedrzejewski** [comment]
Thanks Mateusz!

**Pull request merged**